### PR TITLE
common: don't calibrate perf timer at static init

### PR DESCRIPTION
**Pull request by jpk68**
Lazily calibrates the performance timer instead of doing so on startup. This effectively removes a 100ms wait from the startup time of binaries, which is especially noticeable when running commands such as `--help`.

Before:
```
Executed in  137.48 millis    fish           external
   usr time  104.19 millis  836.00 micros  103.35 millis
   sys time   11.73 millis  314.00 micros   11.41 millis
```

After:
```
Executed in   15.25 millis    fish           external
   usr time    6.09 millis    0.00 micros    6.09 millis
   sys time    9.09 millis  978.00 micros    8.11 millis
```

(both times listed above are for `./monero-wallet-cli --help`)

### CODE REVIEW

**Review by SNeedlewoods — APPROVED** [review]

**Review by thomasbuilds — APPROVED** [review]

File: src/common/perf_timer.cpp [hunk 1]
```diff
@@ -72,12 +73,21 @@ namespace tools
 #endif
 
 #ifdef __x86_64__
-  uint64_t ticks_per_ns = get_ticks_per_ns();
+  namespace
+  {
+    bool calibrate_ticks_per_ns_async()
+    {
+      std::thread([]{ ticks_to_ns(0); }).detach();
```
Comment (thomasbuilds) [inline src/common/perf_timer.cpp:80]: This thread starts before `main()`, and `monerod --detach` / `monero-wallet-rpc --detach` `fork()` 3-4 ms later while it still holds the guard of `ticks_to_ns()`'s static. The child inherits the guard in "initialising" state with no thread to finish it so its first `ticks_to_ns()` blocks forever. Reproduced: `monerod --detach --log-level 1 --offline` hangs at blockchain init after the first `PERF ----------` line, last syscall `futex(<guard of ticks_to_ns::ticks_per_ns>, FUTEX_WAIT, ..., NULL)`, RPC never bound. Any perf logging in the child triggers it (`--log-level 1` or `set_log 1`, then the next RPC, `add_tx` or `check_tx_inputs`).

**Review by thomasbuilds — CHANGES_REQUESTED** [review]

File: src/common/perf_timer.cpp (hunk 1 above)
Comment (jpk68) [reply src/common/perf_timer.cpp]: Thanks, fixed.

File: src/common/perf_timer.cpp [hunk 2]
```diff
@@ -72,12 +74,32 @@ namespace tools
...
+      ticks_per_ns_value.store(tpns, std::memory_order_release);
+      return tpns;
+    }
+
+    bool calibrate_ticks_per_ns_async()
+    {
+      std::thread(calibrate_ticks_per_ns).detach();
+      return true;
+    }
+    const bool ticks_per_ns_calibration_started = calibrate_ticks_per_ns_async();
```
Comment (thomasbuilds) [inline src/common/perf_timer.cpp]: Constructing the thread throws `std::system_error` if it cannot be created, which aborts every binary during static initialization. `ticks_to_ns()` already calibrates on first use, so this block can go:

```suggestion
```

File: src/common/perf_timer.cpp [hunk 3]
```diff
@@ -27,6 +27,8 @@
 // THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
 #include <vector>
+#include <thread>
```
Comment (thomasbuilds) [inline src/common/perf_timer.cpp]:
```suggestion
```

**Review by thomasbuilds — CHANGES_REQUESTED** [review]
Constructing a `std::thread` in a static initializer throws when the thread cannot be created (`EAGAIN`), and an exception escaping a static initializer calls `std::terminate` before `main` is reached:

```
$ prlimit --nproc=1 ./monerod --version   # as a non-root user
terminate called after throwing an instance of 'std::system_error'
  what():  Resource temporarily unavailable
```

I suggest dropping the thread and keeping the lazy path. The startup gain is unchanged, nothing can fail during static initialization, and processes no longer spin for 100 ms in the background at every start. The only cost is a single 100 ms spin in the first thread that needs a conversion, which a detached child already pays in this revision. Concurrent first callers each spin once (8 threads: 101 ms, identical results), so no wait flag is needed; one would reintroduce the `--detach` hang.

**Review by thomasbuilds — APPROVED** [review]

**Review by selsta — APPROVED** [review]

**Review by thomasbuilds — APPROVED** [review]
